packages/fether-react/src/Send/TxForm/TxForm.js

```diff
@@ -303,10 +310,14 @@ class TxForm extends Component {
                               : this.showDetailsAnchor()}
                           </div>
                           <button
-                            disabled={!valid || validating}
+                            disabled={
+                              !valid ||
+                              validating ||
+                              !this.isEstimatedTxFee(values)
+                            }
                             className='button'
                           >
-                            {validating
+                            {validating || !this.isEstimatedTxFee(values)
```

Comment:
This makes the button appear as "Checking..." when Fether isn't actually checking anything (e.g when no field is filled). This is confusing Ux-wise I think.

Reply:
Let's keep the validation logic inside validate()

```js
// If the gas hasn't been calculated yet, then we don't show any errors,
// just wait a bit more
if (!this.estimatedTxFee(values)) {
  return "Estimating gas...";
}
```

does the trick on my side

Reply:
Let's keep the validation logic inside validate()

```js

// If the gas hasn't been calculated yet, then we don't show any errors,
// just wait a bit more
if (!this.isEstimatedTxFee(values)) {
  return "Estimating gas...";
}
```

does the trick for me

Reply:
I'm not sure why, but it only displays the validation popup "Estimating gas..." if you've changed the slider value of the "Transaction Speed"
<img width="348" alt="screen shot 2019-01-24 at 8 16 03 am" src="https://user-images.githubusercontent.com/6226175/51637641-74746100-1fb0-11e9-9bea-b7695647c3ed.png">